test/unit/workflows/test_workflow_progress.py

```diff
@@ -76,7 +76,7 @@ def _setup_workflow(self, workflow_yaml):
         self.invocation.workflow = workflow
 
     def _new_workflow_progress(self):
-        return WorkflowProgress(self.invocation, self.inputs_by_step_id, MockModuleInjector(self.progress), {})
+        return WorkflowProgress(self.invocation, self.inputs_by_step_id, MockModuleInjector(self.progress), {})  # type: ignore[arg-type]
```

Comment:
I love all of it up until here 😅 - can you make it auto-define a protocol? I'm sure that is a big ask. This is fine because type issues in unit test code are either irrelevant or immediately problematic.

Reply:
Is https://github.com/galaxyproject/galaxy/pull/14683/commits/9d802d1551d9c0e50b41ffa335bb1575d5cfce43 what you had in mind ?

Reply:
Very nice - I assumed that would be more work. Wonderful!